flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1191,8 +1197,32 @@ protected int getNumberRequiredTaskManagerSlots() {
 	//  Helper methods
 	// ------------------------------------------------------------------------
 
-	protected static Collection<ResourceProfile> createSlotsPerWorker(int numSlots) {
-		return Collections.nCopies(numSlots, ResourceProfile.ANY);
+	@VisibleForTesting
+	public static Collection<ResourceProfile> createSlotsPerWorker(Configuration config, long totalMemoryMB, int numSlots) {
+		final long cutoffMB = ContaineredTaskManagerParameters.calculateCutoffMB(config, totalMemoryMB);
+
+		final long networkMB = NettyShuffleEnvironmentConfiguration.calculateNetworkBufferMemory(
```

Comment:
In reality `NettyShuffleEnvironmentConfiguration.calculateNetworkBufferMemory` should not be exposed outside based on pluggable shuffle service proposed in [FLINK-12110](https://issues.apache.org/jira/browse/FLINK-12110), so I suggest not covering network factor in `ResourceProfile` atm.

Reply:
I agree that we do not need to cover network memory in `ResourceProfile` in this version. However, I'm afraid we need to access `NettyShuffleEnvironmentConfiguration.calculateNetworkBufferMemory` on RM side anyway. Otherwise, we can not calculate managed memory size from the total container memory size. I think in the future we need a dedicated data structure to unify all the task executor resource calculations, exposing accesses of individual factors while hiding the detailed calculation steps.